core/sql/optimizer/OptRange.cpp

```diff
@@ -503,7 +503,13 @@ ConstValue* OptRangeSpec::getConstOperand(ItemExpr* predExpr, Lng32 constInx)
   // currently support. Predicates involving types not yet supported will be
   // treated as residual predicates.
   if (QRDescGenerator::typeSupported(static_cast<ConstValue*>(right)->getType()))
+  {
```

Comment:
This adds a case-insensitive flag to the type in the RangeSpec, but I don't think RangeSpecs are written to handle case-insensitive comparisons. Take a look at the methods that deal with comparisons when building RangeSpecs, in file Range.cpp. So, I think you would have to do one of two things: a) disable the RangeSpec transformation for case-insensitive comparison operators (the easy way) or change the RangeSpec methods to handle case-insensitive comparisons.

Reply:
I do those operators in order to add case-insensitive to new constvalue.
i don't understand the a
for b, case-insensitive only use in 
             if (parentOfStart)
                parentOfStart->child(1) =
                    new(mvqrHeap_) ConstValue(charSubrange->start,
 if add a RangeSpec methods about case-insensitive , we need add this attribute to parentOfStart->child(1) after operating above.
so i don't think this is a good way

Reply:
RangeSpecs keep a list of points and intervals and these intervals need to be ordered, otherwise RangeSpec doesn't work. If we have case-insensitive comparison, that ordering needs to be different, but as far as I could tell, there is no code at the moment that would do a case-insensitive ordering of the values.